internal/policy/targets.go

```diff
@@ -101,6 +101,16 @@ func AddKeyToTargets(targetsMetadata *tuf.TargetsMetadata, authorizedKeys []*tuf
 	return targetsMetadata, nil
 }
 
+func RemoveKeysFromTargets(targetsMetadata *tuf.TargetsMetadata, authorizedKeys []*tuf.Key) (*tuf.TargetsMetadata, error) {
+	for _, key := range authorizedKeys {
+		if err := targetsMetadata.Delegations.RemoveKey(key.KeyID); err != nil {
```

Comment:
The reason we don't currently do this when we remove a rule is because we need to check another rule doesn't require a key. I recommend having guard rails to stop a user from removing a key that's required by a rule.

Reply:
There is now a check to see if the key is being used by 2 or more rules, including the rules the key is being removed from. Is this what you were thinking about.

Reply:
How should key revocation work if a key is used in multiple rules?